cmd/minishift/cmd/start_preflight.go

```diff
@@ -209,8 +209,20 @@ func checkKvmDriver() bool {
 	if err != nil {
 		return false
 	}
-	fmt.Printf(fmt.Sprintf("\n   Driver is available at %s ... ", path))
+	fi, _ := os.Stat(path)
+	// follow symlinks
+	if fi.Mode()&os.ModeSymlink != 0 {
+		path, err = os.Readlink(path)
+		if err != nil {
+			return false
+		}
+	}
+	fmt.Println(fmt.Sprintf("\n   Driver is available at %s ... ", path))
 
+	fmt.Printf("   Checking for execute bit ...")
+	if fi.Mode()&0011 == 0 {
```

Comment:
I am not sure about this. Any pointers to understand this better.

Reply:
https://golang.org/pkg/os/#FileMode

Reply:
@gbraad I am not sure how `fi.Mode()&0011 ` give us the correct result i.e.fi.Mode() returns  -rwxr-xr-x and we are doing a `&` operation with `0011`. I found https://groups.google.com/forum/#!topic/golang-nuts/_6YqjJdfYyA but still do not understand completely.

Reply:
it is masking the bits for ModePerm. some libraries like go-permbits are a little bit more intuitive. but it is just bitshifted and OR-ed. it checks that o+x and g+x is set.

https://golang.org/src/os/types.go#L35